crates/ra_hir/src/ty.rs

```diff
@@ -529,15 +529,40 @@ struct InferenceContext<'a, D: HirDatabase> {
 
 // helper function that determines whether a binary operator
 // always returns a boolean
-fn is_boolean_operator(op: BinaryOp) -> bool {
+fn boolean_op_return_ty(op: BinaryOp, rhs_ty: Ty) -> Ty {
```

Comment:
this should be called `binary_op_return_ty`, right? it's not just for boolean ops. Also, the comment is outdated

Reply:
Yes!